### ISSUE TITLE
keras_utils.get_file does not validate provided hashes unless file already exists

### ISSUE THREAD

**Issue by moyix** [keras-team/keras#12290]
I was just looking through how Keras downloads and validates datasets and pretrained models. From what I can see everything goes through `keras_utils.get_file`, which can be passed a hash and a URL. But looking at the code it looks like the only time this hash is validated is if there is a pre-existing file with the same name:

https://github.com/keras-team/keras/blob/master/keras/utils/data_utils.py#L196-L202

In other words, for freshly downloaded models, corrupted downloads (or, I suppose, maliciously replaced models) will not be detected. I checked this by changing the hash for the VGG-19 model in `keras-applications` to all zeros; Keras downloaded and loaded it without complaint.

Is this intended behavior? I'd be happy to provide a PR to make `get_file` check the hash after downloading.
